api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -547,4 +547,26 @@ private JSONObject createNotebookCodeCellWithString(String cellInformation) {
         .put("outputs", new JSONArray())
         .put("source", new JSONArray().put(cellInformation));
   }
+
+  @Override
+  public ResponseEntity<EmptyResponse> abortGenomicExtractionJob(
+      String workspaceNamespace, String workspaceId, String jobId) {
+    workspaceAuthService.getWorkspaceEnforceAccessLevelAndSetCdrVersion(
+        workspaceNamespace, workspaceId, WorkspaceAccessLevel.WRITER);
+
+    List<GenomicExtractionJob> allWorkspaceExtractJobs =
+        genomicExtractionService.getGenomicExtractionJobs(workspaceNamespace, workspaceId);
+
+    if (allWorkspaceExtractJobs.stream()
+        .noneMatch(job -> job.getGenomicExtractionJobId().equals(Long.valueOf(jobId)))) {
+      throw new ServerErrorException("Could not find this job in this workspace");
```

Comment:
nit: may be moot per above comment, but 404 would make more sense here (rather than 500)